src/cpp/src/visual_language/minicpm/classes.cpp

```diff
@@ -413,7 +413,31 @@ EncodedImage VisionEncoderMiniCPM::encode(const ov::Tensor& image, const ov::Any
     ctx_clip.image_size = config.image_size;
     std::copy(config.norm_mean.begin(), config.norm_mean.end(), ctx_clip.image_mean);
     std::copy(config.norm_std.begin(), config.norm_std.end(), ctx_clip.image_std);
-    return llava_image_embed_make_with_bytes_slice(ctx_clip, image, encoder, config.max_slice_nums, config.scale_resolution, config.patch_size, 0 == config.max_slice_nums);
+    EncodedImage encoded_image = llava_image_embed_make_with_bytes_slice(ctx_clip, image, encoder, config.max_slice_nums, config.scale_resolution, config.patch_size, 0 == config.max_slice_nums);
+    encoded_image.resampled_image = resample_encoded_image(encoded_image);
+    return encoded_image;
+}
+
+ResampledImage VisionEncoderMiniCPM::resample_encoded_image(const EncodedImage& encoded_image) {
+    const ov::Tensor& resampled_source = resample(encoded_image.resized_source, {encoded_image.resized_source_size});
+    std::vector<std::vector<ov::Tensor>> vision_embed_tensor;
+    if (encoded_image.slices) {
+        size_t token_idx = 0;
+        const ov::Shape& slices_shape = encoded_image.slices.get_shape();
+        vision_embed_tensor.resize(slices_shape.at(0));
+        for (size_t i = 0; i < slices_shape.at(0); ++i) {
+            std::vector<ov::Tensor> vision_embeds;
+            vision_embeds.resize(slices_shape.at(1));
+            for (size_t ja = 0; ja < slices_shape.at(1); ++ja) {
+                size_t d2 = slices_shape.at(2);
+                size_t d3 = slices_shape.at(3);
+                ov::Tensor encoded_view{ov::element::f32, {1, d2, d3}, encoded_image.slices.data<float>() + (i * slices_shape.at(1) + ja) * d2 * d3};
+                vision_embeds[ja] = resample(encoded_view, {encoded_image.slices_size});
```

Comment:
That's the only usage of `slices_size`. `slices_size` member becomes internal to `VisionEncoderMiniCPM` and can be moved out of `EncodedImage`. The same is probably true for `slices` if you replace `if (encoded_image.slices)` with `vision_embed_tensor`. So it's possible to reduce the size of `EncodedImage` although I'm not sure it should be done. Maybe it's simpler to store everything in `EncodedImage` like it's done now.

This also relates to your question about storing the results for models other than minicpm resampler. I think you can stick to naive approach adding new members for new models and leaving comments with your findings which architectures use any given member. After all three architectures are updated, you can collapse `EncodedImage` if it would make sense.

Reply:
Okay, left as it is for now.